app/src/main/java/org/schabi/newpipe/player/MainVideoPlayer.java

```diff
@@ -992,12 +1042,31 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
                 getWindow().setAttributes(lp);
                 if (DEBUG) Log.d(TAG, "onScroll().brightnessControl, currentBrightness = " + currentBrightness);
                 int brightnessNormalized = Math.round(currentBrightness * 100);
+                animateProgressBar(
+                        playerImpl.getBrightnessProgressBar(),
+                        (brightnessNormalized == 1 ? 0 : brightnessNormalized),
+                        100
+                );
+                if (brightnessNormalized < 25) {
+                    playerImpl.getBrightnessImageView().setImageDrawable(
```

Comment:
Crash in old versions of Android.
A very simple solution, thankfully: use `AppCompatResources.getDrawable` instead.

I was also worried about performance, but [it seems to be cached](https://chromium.googlesource.com/android_tools/+/refs/heads/master/sdk/sources/android-25/android/support/v7/widget/AppCompatDrawableManager.java#180), both the Android and compat library versions, so it's all good.

Reply:
Thanks for that - AS stopped complaining when I used it through getResources(), but yeah, I didn't check it manually...